34-cellular-modbus-client/README.md

```diff
@@ -0,0 +1,427 @@
+    - [Notecarrier](#notecarrier)
+  - [Notehub Setup](#notehub-setup)
+  - [Firmware](#firmware)
+    - [Operation](#operation)
+        - [Request Types](#request-types)
+            - [Read Coils](#read-coils)
+            - [Read Discrete Inputs](#read-discrete-inputs)
+            - [Read Holding Registers](#read-holding-registers)
+            - [Read Input Registers](#read-input-registers)
+            - [Write Single Coil](#write-single-coil)
+            - [Write Single Register](#write-single-register)
+            - [Write Multiple Coils](#write-multiple-coils)
+            - [Write Multiple Registers](#write-multiple-registers)
+        - [Response Types](#response-types)
+            - [Reads](#reads)
+            - [Writes](#writes)
+            - [Errors](#errors)
+  - [Testing](#testing)
+    - [Simple Test](#simple-test)
+    - [Test Script](#test-script)
+  - [Blues Community](#blues-community)
+  - [Additional Resources](#additional-resources)
+
+## You Will Need
+
+* [Visual Studio Code (VS Code)](https://code.visualstudio.com/) with the [PlatformIO extension](https://platformio.org/)
+* [Notecarrier F](https://blues.io/products/notecarrier/notecarrier-f/)
+* [Notecard](https://blues.io/products/notecard/)
+* [SparkFun Transceiver Breakout - RS-485](https://www.sparkfun.com/products/10124)
+* [SparkFun USB to RS-485 Converter](https://www.sparkfun.com/products/9822)
+* [2 Screw Terminals 3.5mm Pitch (3-Pin)](https://www.sparkfun.com/products/8235)
+* [STLINK-V3MINI](https://shop.blues.io/collections/accessories/products/stlink-v3mini)
+* 2 micro USB cables
+* [1 USB mini-B cable](https://www.sparkfun.com/products/13243)
+* Male to male jumper wires
+* Female to male jumper wires
+
+## Hardware Setup
+
+This application uses [Modbus RTU](https://en.wikipedia.org/wiki/Modbus#Modbus_RTU_frame_format) with RS-485 for Modbus communication between the client and server. The RS-485 transceiver breakout is used by the client running on the Swan to send and receive Modbus messages. The USB to RS-485 converter is used by the server running on your development PC to send and receive Modbus messages.
+
+### RS-485 Transceiver Breakout
+
+To use the RS-485 transceiver breakout board, solder on a row of 5 male headers and one of the 3-pin screw terminals:
+
+![RS-485 transceiver breakout soldered](images/rs485_transceiver_breakout_soldered.jpg)
+
+Using a small flat head screwdriver, loosen the screws on the screw terminal and insert a male to male jumper wire into each port, tightening the screws after. Then, attach the female end of 5 female to male jumper wires to the male headers you just soldered on:
+
+![RS-485 transceiver breakout wires](images/rs485_transceiver_breakout_wires.jpg)
+
+### USB to RS-485 Converter
+
+To use the USB to RS-485 converter, solder on the remaining 3-pin screw terminal:
+
+![USB to RS-485 converter soldered](images/usb_to_rs485_converter_soldered.jpg)
```

Comment:
Could trim this one too. Only mentioning this because the tall images do make the instructions a bit harder to read/follow.